pkg/server/profiler/profiler_common.go

```diff
@@ -66,26 +91,28 @@ func (o *profiler) maybeTakeProfile(
 	thresholdValue int64,
 	takeProfileFn func(ctx context.Context, path string) bool,
 ) {
-	if resetHighWaterMarkInterval == 0 {
+	if o.resetInterval() == 0 {
 		// Instruction to disable.
 		return
 	}
 
 	now := o.now()
-	// If it's been too long since we took a profile, make sure we'll take one now.
-	if now.Sub(o.lastProfileTime) >= resetHighWaterMarkInterval {
-		o.highwaterMarkBytes = 0
+	// Check whether to reset the high watermark to the floor. This is the case if
+	// the floor (which might change) is now below the high water mark, or if
```

Comment:
I got this wrong in the comment - water mark below floor is when we need to reset. Not the other way around.